cmake_modules/ThirdpartyToolchain.cmake

```
@@ -132,6 +154,9 @@ else ()
     BUILD_BYPRODUCTS "${SNAPPY_STATIC_LIB}")
 
   add_built_library (snappy_ep orc_snappy ${SNAPPY_STATIC_LIB} ${SNAPPY_INCLUDE_DIR})
+
+  list (APPEND ORC_VENDOR_DEPENDENCIES "vendored_snappy:${SNAPPY_STATIC_LIB_NAME}")
+  list (APPEND ORC_INSTALL_INTERFACE_TARGETS "$<INSTALL_INTERFACE:vendored_snappy>")
```

How about putting this target to `orc::` namespace for safety?

```suggestion
  list (APPEND ORC_VENDOR_DEPENDENCIES "orc::vendored_snappy:${SNAPPY_STATIC_LIB_NAME}")
  list (APPEND ORC_INSTALL_INTERFACE_TARGETS "$<INSTALL_INTERFACE:orc::vendored_snappy>")
```

---

This will break the logic here: https://github.com/apache/orc/blob/2ae87d2da38a91728046d9c2fe36fb8b15db5690/c%2B%2B/orcConfig.cmake.in#L47-L49

I can rename it to orc_vendored_snappy

---

Ah, how about using `|` instead of `:` as a separator?

---

Fixed. Thanks for your help!